packages/core/test/deterministic.test.ts

```diff
@@ -97,69 +97,56 @@ describe('deterministic machine', () => {
 
     it('should throw an error if not given an event', () => {
       expect(() =>
-        lightMachine.transition(
+        simulate(lightMachine).transition(
           testMachine.resolveStateValue('red'),
-          undefined as any,
-          undefined as any // TODO: figure out the simulation API
+          undefined as any
         )
       ).toThrow();
     });
 
     it('should transition to nested states as target', () => {
       expect(
-        testMachine.transition(
-          testMachine.resolveStateValue('a'),
-          { type: 'T' },
-          undefined as any // TODO: figure out the simulation API
-        ).value
+        simulate(testMachine).transition(testMachine.resolveStateValue('a'), {
+          type: 'T'
+        }).value
       ).toEqual({
         b: 'b1'
       });
     });
 
     it('should throw an error for transitions from invalid states', () => {
       expect(() =>
-        testMachine.transition(
+        simulate(testMachine).transition(
           testMachine.resolveStateValue('fake'),
-          { type: 'T' },
-          undefined as any // TODO: figure out the simulation API
+          { type: 'T' }
         )
       ).toThrow();
     });
 
     it('should throw an error for transitions from invalid substates', () => {
       expect(() =>
-        testMachine.transition(
+        simulate(testMachine).transition(
           testMachine.resolveStateValue('a.fake'),
           {
             type: 'T'
-          },
-          undefined as any // TODO: figure out the simulation API
+          }
         )
       ).toThrow();
     });
 
     it('should use the machine.initialState when an undefined state is given', () => {
       expect(
-        lightMachine.transition(
-          lightMachine.getInitialState(
-            undefined as any // TODO: figure out the simulation API
-          ),
-          { type: 'TIMER' },
-          undefined as any // TODO: figure out the simulation API
-        ).value
+        simulate(lightMachine).transition(undefined, {
```

Comment:
This `undefined` here looks really suspicious. Im reviewing on mobile so i have a hard time rechecking this across files but it feels like this shouldnt be allowed

Reply:
"Sugar" for initial state

Reply:
I'm not super fond of this - partially because this moves the `input` argument from the `simulate.getInitialState` to the `simulate` itself. This feels weird/out of place because it diverges from the "core" API in an unexpected way.

If this sugar wouldn't be here then `input` could be removed from the `simulate` and there would be a nice "symmetry" between those:
```ts
machine.getInitialState(actorContext, input)
simulate(machine).getInitialState(input)
```

The only difference here would be the implied/provided actor context but other than that the API would stay almost the same.

Reply:
Removed the sugar